django/db/migrations/state.py

```diff
@@ -406,9 +402,9 @@ def update_model_field_relation(
             remote_model_key = concretes[remote_model_key]
         relations_to_remote_model = self._relations[remote_model_key]
         if field_name in self.models[model_key].fields:
-            relations_to_remote_model[model_key].append((field_name, field))
+            relations_to_remote_model[model_key][field_name] = field
```

Comment:
Can you `assert` before this line that `field_name` isn't in `relations_to_remote_model[model_key]`? Before the change, it looks like that would have always been true (though I'm not sure why). After the change, it looks like you might also be replacing a field.

Reply:
> Before the change, it looks like that would have always been true (though I'm not sure why).

Field with the same name could exist only in `alter_field()` operation when the previous one is also relational, however we remove it:
https://github.com/django/django/blob/196a99da5d9c4c33a78259a58d38fb114a4d2ee8/django/db/migrations/state.py#L242-L243

Reply:
Okay, then I would also add a comment mentioning that so people can know why the assert holds.